### ISSUE TITLE
Custom classes inheriting from `ConstantClass` have a private constructor

### ISSUE THREAD

**Issue by bmassemin** [aws/aws-sdk-net#3158]
### Describe the bug

The custom classes inheriting from `ConstantClass` have a private constructor, so the constructor variable is null at this point: https://github.com/aws/aws-sdk-net/blob/6d2e0017c3cf4aa47bd925863342e8c9f49c0be4/sdk/src/Core/Amazon.Runtime/ConstantClass.cs#L108 since `GetConstructor()` by default can only fetch public constructors.

### Expected Behavior

No exception

### Current Behavior

Null Exception due to the constructor variable being null.

### Reproduction Steps

with the s3 client for instance:

```cs
await s3.PutBucketIntelligentTieringConfigurationAsync(new PutBucketIntelligentTieringConfigurationRequest
{
    BucketName = bucketName,
    IntelligentTieringId = Id,
    IntelligentTieringConfiguration = new IntelligentTieringConfiguration
    {
        IntelligentTieringId = Id,
        Status = IntelligentTieringStatus.Enabled,
        Tierings =
        [
            new Tiering
            {
                AccessTier = "SOME_RANDOM_VALUE",
                Days = 999
            }
        ]
    }
});
```

`SOME_RANDOM_VALUE` is not present in the SDK, but `FindValue<T>` should be able to add this value dynamically.

### Possible Solution

Change the constructor to public. The generated classes inheriting from `ConstantClass` have a public constructor in the template.

### AWS .NET SDK and/or Package version used

AWSSDK.S3 3.7.305.15

### Targeted .NET Platform

net8.0

### Operating System and version

Windows 11

**Comment by normj** [comment]
Thanks for pointing out the issue. I put out a PR to address the issue. https://github.com/aws/aws-sdk-net/pull/3159

**Comment by normj** [comment]
PR is merged and should go out with the next release.

**Comment by normj** [comment]
Today's release included the fix in version 3.7.305.17 of AWSSDK.S3. Thanks again for reporting the issue.

https://github.com/aws/aws-sdk-net/blob/main/changelogs/SDK.CHANGELOG.2024.md#377340-2024-01-25-1924-utc
